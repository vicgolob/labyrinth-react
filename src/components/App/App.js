import {useState} from 'react';
import './App.css';
import {MazeEditor} from '../MazeEditor/MazeEditor';
import {MazeGame} from '../MazeGame/MazeGame';

const {pow} = Math;

function App() {
  const [isEditionMode, setEditionMode] = useState(true);
  const [gridSize, setGrid] = useState(4);
  const [mazeSerial, setMazeSerial] = useState('P'.repeat(pow(gridSize, 2)));
  const [{x, y}, setInitialPosition] = useState({x: 0, y: 0});
  const [{endX, endY}, setEndPosition] = useState({endX: 0, endY: 0});
  const [moves, setMoves] = useState(3);
  const [minMoves, setMinMoves] = useState('');

  const showEditor = () => {
    if (isEditionMode) {
      return < MazeEditor
        setMazeSerial={setMazeSerial}
        gridSize={gridSize}
        mazeSerial={mazeSerial}
        setGrid={setGrid}
        setEditionMode={setEditionMode}
        setInitialPosition={setInitialPosition}
        setEndPosition={setEndPosition}
        setMoves={setMoves}
        setMinMoves={setMinMoves}
        setMoves={setMoves}
        moves={moves}
      />;
    } else {
      return < MazeGame
        mazeSerial={mazeSerial}
        gridSize={gridSize}
        initialPosition={{x, y}}
        endPosition={{endX, endY}}
        moves={moves}
        minMoves={minMoves}
        setEditionMode={setEditionMode}
      />;
    }
  };
  return showEditor();
}

export default App;
