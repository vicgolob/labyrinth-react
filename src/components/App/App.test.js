import {render} from '@testing-library/react';
// eslint-disable-next-line no-unused-vars
import App from './App';

describe('App Component', () => (
  it('should render the MazeEditor component', () => {
    const component = render(<App />);
    expect(component).toMatchSnapshot();
  })

));
