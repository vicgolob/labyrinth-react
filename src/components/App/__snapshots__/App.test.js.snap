// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex;"
      >
        <div
          style="display: flex; flex-direction: column;"
        >
          <div
            style="display: inline-flex;"
          >
            <p
              style="margin-right: 1em;"
            >
              Choose grid size
            </p>
            <input
              max="15"
              min="4"
              step="1"
              type="number"
              value="4"
            />
          </div>
          <div
            style="display: inline-flex;"
          >
            <p
              style="margin-right: 1em;"
            >
              Pick to set start position
            </p>
            <div
              style="height: 18px; width: 18px; border-radius: 50%; border: 2px solid black; background-color: rgb(63, 240, 63);"
            />
          </div>
          <div
            style="display: inline-flex;"
          >
            <p
              style="margin-right: 1em;"
            >
              Pick to set end position
            </p>
            <div
              style="height: 18px; width: 18px; border-radius: 50%; border: 2px solid black; background-color: rgb(240, 48, 48);"
            />
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <p
              style="margin-right: 1em;"
            >
              Click the button below to start
            </p>
            <button>
              PLAY ;)
            </button>
          </div>
        </div>
        <table
          cellspacing="0"
          style="padding: 20px;"
        >
          <tbody>
            <tr>
              <td
                id="0"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="1"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="2"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="3"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
            </tr>
            <tr>
              <td
                id="4"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="5"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="6"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="7"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
            </tr>
            <tr>
              <td
                id="8"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="9"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="10"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="11"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
            </tr>
            <tr>
              <td
                id="12"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="13"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="14"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="15"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex;"
    >
      <div
        style="display: flex; flex-direction: column;"
      >
        <div
          style="display: inline-flex;"
        >
          <p
            style="margin-right: 1em;"
          >
            Choose grid size
          </p>
          <input
            max="15"
            min="4"
            step="1"
            type="number"
            value="4"
          />
        </div>
        <div
          style="display: inline-flex;"
        >
          <p
            style="margin-right: 1em;"
          >
            Pick to set start position
          </p>
          <div
            style="height: 18px; width: 18px; border-radius: 50%; border: 2px solid black; background-color: rgb(63, 240, 63);"
          />
        </div>
        <div
          style="display: inline-flex;"
        >
          <p
            style="margin-right: 1em;"
          >
            Pick to set end position
          </p>
          <div
            style="height: 18px; width: 18px; border-radius: 50%; border: 2px solid black; background-color: rgb(240, 48, 48);"
          />
        </div>
        <div
          style="display: flex; flex-direction: column;"
        >
          <p
            style="margin-right: 1em;"
          >
            Click the button below to start
          </p>
          <button>
            PLAY ;)
          </button>
        </div>
      </div>
      <table
        cellspacing="0"
        style="padding: 20px;"
      >
        <tbody>
          <tr>
            <td
              id="0"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="1"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="2"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="3"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
          </tr>
          <tr>
            <td
              id="4"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="5"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="6"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="7"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
          </tr>
          <tr>
            <td
              id="8"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="9"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="10"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="11"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
          </tr>
          <tr>
            <td
              id="12"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="13"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="14"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="15"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App Component should render the MazeEditor component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex;"
      >
        <div
          style="display: flex; flex-direction: column;"
        >
          <div
            style="display: inline-flex;"
          >
            <p
              style="margin-right: 1em;"
            >
              Choose grid size
            </p>
            <input
              max="15"
              min="4"
              step="1"
              type="number"
              value="4"
            />
          </div>
          <div
            style="display: inline-flex;"
          >
            <p
              style="margin-right: 1em;"
            >
              Pick to set start position
            </p>
            <div
              style="height: 18px; width: 18px; border-radius: 50%; border: 2px solid black; background-color: rgb(63, 240, 63);"
            />
          </div>
          <div
            style="display: inline-flex;"
          >
            <p
              style="margin-right: 1em;"
            >
              Pick to set end position
            </p>
            <div
              style="height: 18px; width: 18px; border-radius: 50%; border: 2px solid black; background-color: rgb(240, 48, 48);"
            />
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <p
              style="margin-right: 1em;"
            >
              Click the button below to start
            </p>
            <button>
              PLAY ;)
            </button>
          </div>
        </div>
        <table
          cellspacing="0"
          style="padding: 20px;"
        >
          <tbody>
            <tr>
              <td
                id="0"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="1"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="2"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="3"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
            </tr>
            <tr>
              <td
                id="4"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="5"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="6"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="7"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
            </tr>
            <tr>
              <td
                id="8"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="9"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="10"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="11"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
            </tr>
            <tr>
              <td
                id="12"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="13"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="14"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
              <td
                id="15"
                style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex;"
    >
      <div
        style="display: flex; flex-direction: column;"
      >
        <div
          style="display: inline-flex;"
        >
          <p
            style="margin-right: 1em;"
          >
            Choose grid size
          </p>
          <input
            max="15"
            min="4"
            step="1"
            type="number"
            value="4"
          />
        </div>
        <div
          style="display: inline-flex;"
        >
          <p
            style="margin-right: 1em;"
          >
            Pick to set start position
          </p>
          <div
            style="height: 18px; width: 18px; border-radius: 50%; border: 2px solid black; background-color: rgb(63, 240, 63);"
          />
        </div>
        <div
          style="display: inline-flex;"
        >
          <p
            style="margin-right: 1em;"
          >
            Pick to set end position
          </p>
          <div
            style="height: 18px; width: 18px; border-radius: 50%; border: 2px solid black; background-color: rgb(240, 48, 48);"
          />
        </div>
        <div
          style="display: flex; flex-direction: column;"
        >
          <p
            style="margin-right: 1em;"
          >
            Click the button below to start
          </p>
          <button>
            PLAY ;)
          </button>
        </div>
      </div>
      <table
        cellspacing="0"
        style="padding: 20px;"
      >
        <tbody>
          <tr>
            <td
              id="0"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="1"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="2"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="3"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
          </tr>
          <tr>
            <td
              id="4"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="5"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="6"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="7"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
          </tr>
          <tr>
            <td
              id="8"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="9"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="10"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="11"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
          </tr>
          <tr>
            <td
              id="12"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="13"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="14"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
            <td
              id="15"
              style="height: 22px; width: 22px; background-color: rgb(244, 243, 243); border: .3px solid #959595;"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
